cmd/main.go

```diff
@@ -72,19 +61,23 @@ var (
 	featureMaxSharePerInstance = flag.Bool("feature-max-shares-per-instance", false, "If this feature flag is enabled, allows the user to configure max shares packed per Filestore instance")
 	descOverrideMaxShareCount  = flag.String("desc-override-max-shares-per-instance", "", "If non-empty, the filestore instance description override is used to configure max share count per instance. This flag is ignored if 'feature-max-shares-per-instance' flag is false. Both 'desc-override-max-shares-per-instance' and 'desc-override-min-shares-size-gb' must be provided. 'ecfsDescription' is ignored, if this flag is provided.")
 	descOverrideMinShareSizeGB = flag.String("desc-override-min-shares-size-gb", "", "If non-empty, the filestore instance description override is used to configure min share size. This flag is ignored if 'feature-max-shares-per-instance' flag is false. Both 'desc-override-max-shares-per-instance' and 'desc-override-min-shares-size-gb' must be provided. 'ecfsDescription' is ignored, if this flag is provided.")
+	coreInformerResyncPeriod   = flag.Duration("core-informer-resync-repriod", 15*time.Minute, "Core informer resync period.")
 
 	// Feature stateful CSI driver specific parameters
-	enableStateful = flag.Bool("enable-stateful-multishare", false, "")
-
-	kubeconfig               = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
-	coreInformerResyncPeriod = flag.Duration("core-informer-resync-repriod", 15*time.Minute, "Core informer resync period.")
+	enableStateful          = flag.Bool("enable-stateful-multishare", false, "")
+	featureStateful         = flag.Bool("feature-stateful-multishare", false, "if set to true, the controller will run stateful multishare controller")
+	resyncPeriod            = flag.Duration("stateful-resync-period", 15*time.Minute, "Resync interval of the stateful driver.")
+	kubeAPIQPS              = flag.Float64("kube-api-qps", 5, "QPS to use while communicating with the kubernetes apiserver. Defaults to 5.0.")
+	kubeAPIBurst            = flag.Int("kube-api-burst", 10, "Burst to use while communicating with the kubernetes apiserver. Defaults to 10.")
+	kubeconfig              = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	leaderElection          = flag.Bool("leader-election", false, "Enables leader election for stateful driver.")
```

Comment:
add some stateful prefix to differentiate from the leader-election used by the lock release controller  L55

Reply:
I don't think there are any difference. I'm reusing the `leaderElectionLeaseDuration`, `leaderElectionRenewDeadline` and `leaderElectionRetryPeriod` above also

Reply:
ok. let's keep the leader election flags definition together

Reply:
done